### DISCUSSION TITLE
VPN Access w/Docker Compose

### DISCUSSION THREAD

**Discussion by metaskills** [community/community#5870] [Codespaces] [answered]
Following up from https://github.com/github/feedback/discussions/5767, I have noticed that when connecting with a VPN that Docker Compose services via their service name will be unreachable. cc @invacuo

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @metaskills, thanks for your feedback. Supporting accessing on-prem/VPN based/cloud based resources is an area we're actively exploring and I've added a response around that here: https://github.com/github/feedback/discussions/5741#discussioncomment-1448503

Let's continue the discussion there :)

**Reply by metaskills** [reply]
Thanks, left a comment there. I think the issue is the local docker network with compose conflicting with our VPN IP range. This vide explains it well and I think the ball is our court to address it. https://www.youtube.com/watch?v=BK8NcHUNF5M

**Reply by metaskills** [reply]
If it helps others find this thread, there are a few ways you can change how Docker services use IP addresses. One that I found really helpful using the `docker-compose.yml` file vs the docker.json would be to override the default network like so. Change the subnet to suit your needs.

```dockerfile
networks:
  default: { ipam: { config: [{ subnet: 192.168.176.0/24 }] } }
```
